File: 06/Java/TripleExpression/ShiftLeft.java
package TripleExpression;

public class ShiftLeft extends BinaryOperator implements TripleExpression {
    public ShiftLeft(TripleExpression first, TripleExpression second) {
        super(first, second);
    }

    protected int apply(int a, int b) {
        return a << b;
    }
}

File: 07/CSharp/TripleExpressionApp/TripleExpressionApp/ShiftLeft.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace TripleExpressionApp
{
    public class ShiftLeft : BinaryOperator
    {
        public ShiftLeft(ITripleExpression first, ITripleExpression second)
        {
            fOp = first;
            sOp = second;
        }

        protected override double Apply(double a, double b)
        {
            return (int)a << (int)b;
        }
    }
}
